terraform/eks/daemon/awsneuron/main.tf

```diff
@@ -0,0 +1,826 @@
+  vars     = {}
+}
+data "template_file" "httpd_ssl_config" {
+  template = file(local.httpd_ssl_config)
+  vars     = {}
+}
+
+resource "kubernetes_config_map" "httpdconfig" {
+  depends_on = [
+    kubernetes_namespace.namespace,
+    kubernetes_service_account.cwagentservice
+  ]
+  metadata {
+    name      = "httpdconfig"
+    namespace = "amazon-cloudwatch"
+  }
+  data = {
+    "httpd.conf" : data.template_file.httpd_config.rendered
+    "httpd-ssl.conf" : data.template_file.httpd_ssl_config.rendered
+  }
+}
+
+resource "kubernetes_service_account" "cwagentservice" {
+  depends_on = [kubernetes_namespace.namespace]
+  metadata {
+    name      = "cloudwatch-agent"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "kubernetes_cluster_role" "clusterrole" {
+  depends_on = [kubernetes_namespace.namespace]
+  metadata {
+    name = "cloudwatch-agent-role"
+  }
+  rule {
+    verbs      = ["get", "list", "watch"]
+    resources  = ["pods", "pods/logs", "nodes", "nodes/proxy", "namespaces", "endpoints"]
+    api_groups = [""]
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["replicasets"]
+    api_groups = ["apps"]
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["jobs"]
+    api_groups = ["batch"]
+  }
+  rule {
+    verbs      = ["get"]
+    resources  = ["nodes/proxy"]
+    api_groups = [""]
+  }
+  rule {
+    verbs      = ["create"]
+    resources  = ["nodes/stats", "configmaps", "events"]
+    api_groups = [""]
+  }
+  rule {
+    verbs          = ["get", "update"]
+    resource_names = ["cwagent-clusterleader"]
+    resources      = ["configmaps"]
+    api_groups     = [""]
+  }
+  rule {
+    verbs          = ["get"]
+    resource_names = ["neuron-monitor-config-map"]
+    resources      = ["configmaps"]
+    api_groups     = [""]
+  }
+  rule {
+    verbs      = ["list", "watch"]
+    resources  = ["services"]
+    api_groups = [""]
+  }
+  rule {
+    non_resource_urls = ["/metrics"]
+    verbs             = ["get", "list", "watch"]
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "rolebinding" {
+  depends_on = [kubernetes_namespace.namespace]
+  metadata {
+    name = "cloudwatch-agent-role-binding"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "cloudwatch-agent-role"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "cloudwatch-agent"
+    namespace = "amazon-cloudwatch"
+  }
+}
+
+resource "null_resource" "validator" {
+  depends_on = [
+    aws_eks_node_group.this,
+    kubernetes_daemonset.service,
+    kubernetes_cluster_role_binding.rolebinding,
+    kubernetes_service_account.cwagentservice,
+  ]
+  provisioner "local-exec" {
+    command = <<-EOT
+      echo "Validating EKS metrics/logs for EMF"
+      cd ../../../..
+      go test -timeout 30m ${var.test_dir} -eksClusterName=${aws_eks_cluster.this.name} -computeType=EKS -v -eksDeploymentStrategy=DAEMON -eksGpuType=nvidia
```

Comment:
it might not be relevant to this specific test but we can probably drop this gpu type if is not needed for other types.

Reply:
removed